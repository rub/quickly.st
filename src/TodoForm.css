.todo_form .form {
  position: relative;
}

.todo_form .input {
  background: none;
  border: none;
  border-bottom: 1px solid #A4ADB6;
  color: #FFF;
  font-family: Roboto;
  font-size: 1rem;
  padding: 0.382rem 0;
  width: 100%;
}

.todo_form .input::placeholder {
  color: #A4ADB6;
}

.todo_form .input:focus {
  outline: none;
}

.todo_form .add_container {
  bottom: 0;
  position: absolute;
  right: 0;
}

/**
 * A small trick to use the SVG as a submit button. First, we wrap both the
 * submit button and the SVG inside a <label> so that they are associated
 * together, then we hide the submit button.
 */
.hidden_submit {
  display: none;
}

.todo_form .is_not_empty > svg > path {
  fill: #58E1CC;
}
