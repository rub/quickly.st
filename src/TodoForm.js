import React from 'react';


class TodoForm extends React.Component {
  render(foo) {
    return (
      <form>
        <input
          placeholder="Add new item"
          onChange={this.props.value}
        />
        <button type="submit">+</button>
      </form>
    )
  }
}

export default TodoForm;
