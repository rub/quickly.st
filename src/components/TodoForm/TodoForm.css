.TodoForm {
  position: relative;
}

.TodoForm-submit {
  bottom: 0.146rem;
  cursor: pointer;
  position: absolute;
  right: 0;
}

/**
 * We need the input submit in order to keep the Enter key press event
 * available.
 */
.TodoForm-submit-hidden {
  display: none;
}

.TodoForm-submit-visible path {
  transition: 0.17s ease;
}

.TodoForm-submit-visible.is_filled path {
  fill: #58E1CC;
}
