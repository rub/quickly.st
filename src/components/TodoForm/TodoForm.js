import React from 'react';
import TodoInput from './TodoInput/TodoInput';
import SvgIcon from '../SvgIcon/SvgIcon';
import PropTypes from 'prop-types';

import './TodoForm.css';

/**
 * The input form where user can create todos
 */
const TodoForm = (props) => {
  const addStyles = props.value !== '' ? 'is_filled' : '';

  return (
    <div className="TodoForm">
      <TodoInput value={props.value} onChange={props.onChange} />
      <label className="TodoForm-submit">
        <input className="TodoForm-submit-hidden" type="submit" />
        <SvgIcon
          className={`TodoForm-submit-visible ${addStyles}`}
          icon="add"
          onSubmit={props.onSubmit}
        />
      </label>
    </div>
  )
}

TodoForm.propTypes = {
  /** TodoInput must have a value */
  value: PropTypes.string.isRequired,
  
  /** TodoInput must have a function which stores the user input */
  onChange: PropTypes.func.isRequired,

  /** TodoForm must have a function which submits the form */
  onSubmit: PropTypes.func,
}

export default TodoForm;
