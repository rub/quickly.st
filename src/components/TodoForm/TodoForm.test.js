import React from 'react';
import TodoForm from './TodoForm';
import renderer from 'react-test-renderer';


it('renders the form without crashing', () => {
  const tree = renderer.create(
    <TodoForm
      value="I'm a todo"
      onChange={jest.fn()}
      onSubmit={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})