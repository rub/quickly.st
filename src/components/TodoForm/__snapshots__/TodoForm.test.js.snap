// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the form without crashing 1`] = `
<div
  className="TodoForm"
>
  <div
    className="TodoInput"
  >
    <input
      autoFocus={true}
      className="TodoInput-input"
      onChange={[Function]}
      placeholder="Add new item"
      type="text"
      value="I'm a todo"
    />
  </div>
  <label
    className="TodoForm-submit"
  >
    <input
      className="TodoForm-submit-hidden"
      type="submit"
    />
    <svg
      className="TodoForm-submit-visible is_filled"
      height="18"
      viewBox="0 0 24 24"
      width="18"
    >
      <path
        d="M9.7 0v9.7H0v4.6h9.7V24h4.6v-9.7H24V9.7h-9.7V0H9.7z"
        fill="#88929F"
      />
    </svg>
  </label>
</div>
`;
